src/transformers/models/fast/modeling_fast.py

```diff
@@ -0,0 +1,380 @@
+
+        # TODO: check if needed
+        self.rbr_identity = nn.BatchNorm2d(out_channels) if out_channels == in_channels and stride == 1 else None
+
+    def forward(self, hidden_states: torch.Tensor):
+        # if self.training:
+        main = self.main_conv(hidden_states)
+        main = self.main_batch_norm(main)
+
+        vertical = self.vertical_conv(hidden_states)
+        vertical = self.vertical_batch_norm(vertical)
+
+        horizontal = self.horizontal_conv(hidden_states)
+        horizontal = self.horizontal_batch_norm(horizontal)
+
+        if self.rbr_identity is not None:
+            rbr_identity = self.rbr_identity(hidden_states)
+        else:
+            rbr_identity = 0
+
+        return self.activation(main + vertical + horizontal + rbr_identity)
+
+
+class FastPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = FastConfig
+    base_model_prefix = "fast"
+    main_input_name = "pixel_values"
+
+    def _init_weights(self, module):
+        if isinstance(module, nn.Linear):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Conv2d):
+            module.weight.data.zero_()
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.BatchNorm2d):
+            module.weight.data.fill_(1.0)
+            module.bias.data.zero_()
+
+
+class FASTNeck(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.reduce_layers = nn.ModuleList()
+        for in_channels, out_channels, kernel_size, stride in zip(
+            config.neck_in_channels,
+            config.neck_out_channels,
+            config.neck_kernel_size,
+            config.neck_stride,
+        ):
+            layer = FastRepConvLayer(in_channels, out_channels, kernel_size, stride)
+            self.reduce_layers.append(layer)
+        self.num_layers = len(self.reduce_layers)
+
+    def forward(self, hidden_states: torch.Tensor):
+        first_layer_hidden = hidden_states[0]
+        first_layer_hidden = self.reduce_layers[0](first_layer_hidden)
+        output_stages = [first_layer_hidden]
+
+        for layer_ix in range(1, self.num_layers):
+            layer_out = self.reduce_layers[layer_ix](hidden_states[layer_ix])
+            _, _, height, width = first_layer_hidden.size()
+            layer_out = F.interpolate(layer_out, size=(height, width), mode="bilinear")
+            output_stages.append(layer_out)
+
+        combined_hidden_states = torch.cat(output_stages, 1)
+        return combined_hidden_states
+
+
+class FASTHead(nn.Module):
+    def __init__(self, config: FastConfig) -> None:
+        super().__init__()
```

Comment:
And everywhere, please check consistency

```suggestion
class FastHead(nn.Module):
    def __init__(self, config: FastConfig) -> None:
        super().__init__()
```